examples/miscellaneous/plot_kernel_ridge_regression.py

```diff
@@ -57,19 +54,34 @@
 
 X_plot = np.linspace(0, 5, 100000)[:, None]
 
-# #############################################################################
-# Fit regression model
+# %%
+# Fit the regression models
+# -------------------------
+
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVR
+from sklearn.kernel_ridge import KernelRidge
+
 train_size = 100
+
+# Fit the SVR
 svr = GridSearchCV(
     SVR(kernel="rbf", gamma=0.1),
     param_grid={"C": [1e0, 1e1, 1e2, 1e3], "gamma": np.logspace(-2, 2, 5)},
 )
 
+# Fit the Kernel Ridge Regression
 kr = GridSearchCV(
     KernelRidge(kernel="rbf", gamma=0.1),
     param_grid={"alpha": [1e0, 0.1, 1e-2, 1e-3], "gamma": np.logspace(-2, 2, 5)},
 )
 
+# %%
+# Compare times of SVR and Kernel Ridge Regression
+# ----------------------------------------------
+
+import time
+
 t0 = time.time()
 svr.fit(X[:train_size], y[:train_size])
 svr_fit = time.time() - t0
```

Comment:
In the lines below, it would be interesting to also display the results of the hyper-parameter search for both models, for instance:

```python
print(f"Best SVR with params: {svr.best_params_} and R2 score: {svr.best_score_:.3f}")
print(f"Best KRR with params: {kr.best_params_} and R2 score: {kr.best_score_:.3f}")
```

Then we could reuse those tuned hyper-parameters for the latest figure instead of using arbitrary values for `alpha`, `C` and `gamma`.